distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/catalog/CatalogTestCommons.java

```diff
@@ -299,11 +289,49 @@ public static void deleteMetacard(String id, int expectedStatusCode) {
 
     /**
      * Uses ids within the responses to delete
+     *
      * @param response - response with ids of metacards to delete
      */
     public static void deleteMetacardUsingCswResponseId(Response response)
             throws IOException, XPathExpressionException {
         String id = getMetacardIdFromCswInsertResponse(response);
         CatalogTestCommons.deleteMetacard(id);
     }
+
+    /**
+     * Ingest might not succeed the first time due to the async nature of some configurations.
+     * The ingest service can also return HTTP status code 201 while it creates the metacard
+     * asynchronously. This method invokes a request function that send an HTTP request and
+     * returns a single Metacard ID from the response. The confirmation function is invoked
+     * repeatedly until the condition is met. For an ingest request, the confirmation function
+     * should return true if the Metacard exists in the Catalog. For a delete requests, the
+     * confirmation function should return true if the Metacard does NOT exist in the Catalog.
+     * For an update request, the confirmation function should query the Catalog and ensure the
+     * Metacard's values were changed as expected.
+     * This method will retry the confirmation function several times before giving up
+     * and throwing an exception.
+     *
+     * @param requestFunction a zero-arg function that returns the a Metacard's ID
+     * @return Metacard ID
+     */
+    public static String requestAndConfirm(Supplier<String> requestFunction,
+            Function<String, Boolean> confirmationFunction) {
+        String[] id = new String[1];
```

Comment:
Java 8 lambdas have a limitation where free variables have to be final (or effectively final). The "workaround" was to use an array with one element. The array was effectively final, so you could mutate the element's one array.
IntelliJ refactoring just proposed a different solution to me-- make encapsulate the variable in an atomic reference. I don't know much about them, but I don't see a reason not to use them. It looks more elegant than the 1 element array solution.